dspace-server-webapp/src/main/java/org/dspace/app/rest/model/FilteredItemRest.java

```diff
@@ -0,0 +1,58 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.app.rest.model;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+/**
+ * Specialization of ItemRest dedicated to the Filtered Items report.
+ * This class adds the owning collection property required to properly
+ * display search results without compromising the expected behaviour
+ * of standard ItemRest instances, in all other contexts, especially
+ * when it comes to embedded contents, a criterion that is widely checked
+ * against in several integration tests.
+ *
+ * @author Jean-François Morin (jean-francois.morin@bibl.ulaval.ca)
+ */
+@LinksRest(links = {
+        @LinkRest(
+                name = FilteredItemRest.OWNING_COLLECTION,
+                method = "getOwningCollection"
+        )
+})
+public class FilteredItemRest extends ItemRest {
```

Comment:
I think this might be part of the bug?  It's inheriting from ItemRest.  Maybe you need to make this separate so that it only `implements RestModel`?   I think the backend is expecting this to have a corresponding RestRepository class if you extend `RestAddressableModel` (which this currently does)